### DISCUSSION TITLE
Custom fields and other addons for JoomCCK

### DISCUSSION THREAD

**Discussion by Giorgi625** [JoomCoder-com/JoomCCK#261] [Ideas]
So we see that Andi and his team has lots of work to do, improve JoomCCK and fix bugs, work on releasing JoomSubscription and also they have many other extensions on JoomCoder and JoomBoost, so because of this we have pretty slow support.
Some of you may remember me from mintjoomla where I have published and successfully sold some of my addons for Cobalt.
Last days I was experimenting with JoomCCK and I was able to get some good results.
I am mostly experienced in creating template for JoomCCK, but recently started to try my luck in creating fields and decided to share some preview of my works:
Below you see Working hours field. How it looks on narrow and wide places. in submit form you can set start and end times for each day or set as closed. also you can select Holiday dates. It also check if right now this business is open or closed according to data you have entered in submit form and shows status on top. you can see open has green background. Closed status has red background. Working on improving some features and soon plan to release.
![WhatsApp Image 2024-12-25 at 16 50 01_c24fed23](https://github.com/user-attachments/assets/4fd4b6b2-be30-4f16-afbb-da237846aac4)
![Working hours output](https://github.com/user-attachments/assets/e176a325-e750-4830-b3b2-dff3f5522352)
![Working hours input](https://github.com/user-attachments/assets/ed7e9c9f-6c50-4e21-815c-de80fe3f70e2)

Also I was able to create simple version of subform field, where fields are predefined. in my case most complex version of subform I needed has this structure:
It is used in creation of CV, field called Education, so here you are able to select your education level which values you prepare in field options in textarea.
![image](https://github.com/user-attachments/assets/ec136247-ce86-40a7-bfaa-fda2b994358d)

below you see how it looks in input form, we have select with education levels dropdown, then next input field is to type University, School or such organizations, third field is faculty. then there are two date fields Start and End dates, but there is also checkbox which says Still studying. if you check it second date field gets disabled. and on bottom left is add more button, in case you studied in multiple places you are able to add as many as you wish. 
![image](https://github.com/user-attachments/assets/78807869-5f24-4a4d-820b-3a52845ea435)

Here you can see how it looks in article. 
You see first with big letter Degree, then University, then Faculty and then dates. as you can see on last line there is date and then word, this one is Still studying. 
![image](https://github.com/user-attachments/assets/ec4915d5-31e0-40c2-8f66-cb0b37d7e643)

I am working on project which has only Georgian language so sorry I do not have demo to display it in English at this moment.

Below you see one of my list view templates
![image](https://github.com/user-attachments/assets/d72239bb-a2d2-471c-ae3c-0cd476ca59fe)

This is another template which has some shadows around on hover, soo I will share link to demo.
![image](https://github.com/user-attachments/assets/7fd1cbc6-8597-4720-9871-3b1b3496367b)

**So If you need some custom work about field, templates or other stuff let me know.**  
PS. If you order custom work to me, all the money I get, will be given to Andi to add new features to JoomCCK or to Speed up work on JoomSubscription.

**Comment by KEZEO** [comment]
yes   i remember i bought some customs works to you few years ago for cobalt. i was happy with your work.  joomcck need better design hope you can do something. how to buy this field please ?

**Reply by Giorgi625** [reply]
Thanks, I will soon share website where I will post all my custom works.

**Reply by KEZEO** [reply]
i think now andi could try to do forum template for joomcoder. forum template in cobalt was very popular with his unique feature.
and i think andi and you could try to implement rich snippet / schema in joomcoder theme like blog, forum posting,  q and a , ect
SEO is became very important. thank you

**Comment by KEZEO** [comment]
If you have beautifull blog post template with blog post schema, i will buy too lol

**Comment by CMS-CORP** [comment]
waiting for that too, ready to buy , i need to accelerate for my customers
2 questions  : 1) possilble to have  such theme for joomcck 
![image](https://github.com/user-attachments/assets/d10b24e8-4593-40b9-bf59-6829112dc4c8)
2) i would like to display parent/child or related item in side for full view
![image](https://github.com/user-attachments/assets/1731986d-d82f-44e6-91f0-e5ac1f9c8f4c) (check the sidebar to left)

thanks

**Comment by Giorgi625** [comment]
You can add me on **whatsapp** +995568813977 or **telegram** @giorgibulia and provide details of your requirements and I will try to fulfil your needs as soon as I can.

**Reply by JEROMELECORSE** [reply]
you think you could publish a demo version soon ?

**Reply by Giorgi625** [reply]
I am trying my best to publish it as soon as possible.

**Reply by JEROMELECORSE** [reply]
thank you very much

**Comment by Giorgi625** [comment]
I have published new fields :

- Before / After 
- Polls
- Repeatable text
- Skills
- Gallery with Watermark
- Emotions
- Embed Facebook page

Coming Soon

- Directions field which will let users display direction from one place to another on a google map. 
- Trek field which will allow users to upload their gps trek files and display hiking routes with all details(similar to https://shondalai.com/products/gps-tools/ )

Check Demos here https://joomla.ge/en/demos/fields-demos/item/42

**Reply by andiamine** [reply]
wooow, is better to create discussion of each field you did with screenshot and features list

**Reply by Giorgi625** [reply]
ok I will do so
